helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -798,25 +799,18 @@ private void updateDisabledInstances(Collection<InstanceConfig> instanceConfigs,
     _disabledInstanceSet.clear();
     for (InstanceConfig config : instanceConfigs) {
       Map<String, List<String>> disabledPartitionMap = config.getDisabledPartitionsMap();
-      if (!config.getInstanceEnabled()) {
+      if (!InstanceValidationUtil.isInstanceEnabled(config, clusterConfig)) {
```

Comment:
Why we need to move the logic to the Util?

Reply:
I think it's cleaner to have a unified way to do such check.